lib/server.js

```diff
@@ -844,30 +870,48 @@ WalletService.prototype.createTx = function(opts, cb) {
           return cb(new ClientError('NOTALLOWEDTOCREATETX', 'Cannot create TX proposal during backoff time'));
 
         var copayer = wallet.getCopayer(self.copayerId);
-        var hash = WalletUtils.getProposalHash(opts.toAddress, opts.amount, opts.message, opts.payProUrl);
+        var proposalHeader = Model.TxProposal.create(opts).getHeader();
+        var hash = WalletUtils.getProposalHash.apply(WalletUtils, proposalHeader);
         if (!self._verifySignature(hash, opts.proposalSignature, copayer.requestPubKey))
           return cb(new ClientError('Invalid proposal signature'));
 
-        var toAddress;
-        try {
-          toAddress = new Bitcore.Address(opts.toAddress);
-        } catch (ex) {
+        var outputs = (opts.type == Model.TxProposal.Types.MULTIPLEOUTPUTS)
+          ? opts.outputs
+          : [ { toAddress: opts.toAddress, amount: opts.amount } ];
+        var badAddress = false,
+            badNetwork = false,
+            badAmount = false,
+            badDust = false;
+        _.each(outputs, function(output) {
+          var toAddress;
+          try {
+            toAddress = new Bitcore.Address(output.toAddress);
+          } catch (ex) {
+            badAddress = true;
+          }
+          if (toAddress.network != wallet.getNetworkName())
```

Comment:
If there is an exception in the previous `try catch` block, `toAddress` will be `undefined` here